hosters/hetzner-dedicated/hetzner-dedicated-wipe-and-install-nixos.sh

```diff
@@ -30,13 +24,80 @@
 #   being able to login without any authentication.
 # * The script reboots at the end.
 
-set -eu
-set -o pipefail
 
-set -x
+# Default options
+RAIDLEVEL=1
+HOSTNAME='hetzner'
+NETWORK_BRIDGE=0
+
+# Default SSH key
+SSH_PUBKEY="$(xargs < <(find "$HOME/.ssh" -name '*.pub' -exec cat {} \; -quit))"
+
+# Strict mode
+set -euo pipefail
+
+# Parameter parsing
+while true; do
+  case "$1" in
+    # Construct RAID in striped mode rather than mirror
+    --raid0)
+      RAIDLEVEL=0
+      shift
+      ;;
+
+    # Configure a network bridge and enslave the primary NIC
+    --bridge)
+      NETWORK_BRIDGE=1
+      shift
+      ;;
+
+    # Configure a known hostname
+    --hostname)
+      HOSTNAME="$2"
+      shift 2
+      ;;
+
+    # Provide the path to an SSH public key file to use for the root user
+    --pubkey)
+      SSH_PUBKEY="$(xargs < "$2")"
+      shift 2
+      ;;
+
+    # Print shell statements
+    --verbose|-v)
+      VERBOSE=1
+      shift
+      ;;
+
+    # Print usage information
+    --help|-h)
+      cat <<-EOF
+				usage: ssh root@<hostname> bash -s < $0 [options]
+
+				OPTIONS
+						--raid0                   Construct the root RAID array in striped mode rather than mirrored mode
+						--bridge                  Provision a network bridge and enslave the primary NIC to it
+						--hostname HOSTNAME       Set a known hostname, used for both the host and the madm RAID
+						--pubkey FILE             Provide the path to a file containing an SSH public key to be provisioned to the root user
+						--verbose|-h              Enable shell command tracing
+				EOF
+      ;;
+
+    # Stop parsing option on first unknown parameter
+    *)
+      break
+  esac
+done
+
+[[ -n $VERBOSE ]] && set -x
+
+# Install dependencies
+apt-get update
```

Comment:
I think this might be redundant? it has worked without it in my testing and it shouldn't matter if our copy of `sudo` is up to date

Reply:
Probably, I'll just remove it.

Reply:
Removed